llvm/test/CodeGen/Thumb/PR35481.ll

```diff
@@ -18,11 +18,10 @@ define <4 x i32> @f() local_unnamed_addr #0 {
 ; CHECK-V4T-NEXT:    movs r2, #3
 ; CHECK-V4T-NEXT:    movs r3, #4
 ; CHECK-V4T-NEXT:    bl g
+; CHECK-V4T-NEXT:    ldr r7, [sp, #4]
```

Comment:
This doesn't look right: if we have a frame pointer in r7, we can't use r7 to store the return address.  (This matters for anything trying to asynchronously unwind using the frame pointer chain, like profilers.)

Reply:
r7 is popped to later, so this is fine.

Reply:
In this context, r7 is popped after it is used to restore the return address to lr.

Reply:
Like, immediately after.